src/components/views/settings/KeyBackupPanel.js

```diff
@@ -0,0 +1,240 @@
+    _startNewBackup() {
+        const CreateKeyBackupDialog = sdk.getComponent("dialogs.keybackup.CreateKeyBackupDialog");
+        Modal.createTrackedDialog('Key Backup', 'Key Backup', CreateKeyBackupDialog, {
+            onFinished: () => {
+                this._loadBackupStatus();
+            },
+        });
+    }
+
+    _deleteBackup() {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        Modal.createTrackedDialog('Delete Backup', '', QuestionDialog, {
+            title: _t("Delete Backup"),
+            description: _t(
+                "Delete your backed up encryption keys from the server? " +
+                "You will no longer be able to use your recovery key to read encrypted message history",
+            ),
+            button: _t('Delete backup'),
+            danger: true,
+            onFinished: (proceed) => {
+                if (!proceed) return;
+                this.setState({loading: true});
+                MatrixClientPeg.get().deleteKeyBackupVersion(this.state.backupInfo.version).then(() => {
+                    this._loadBackupStatus();
+                });
+            },
+        });
+    }
+
+    _restoreBackup() {
+        const RestoreKeyBackupDialog = sdk.getComponent("dialogs.keybackup.RestoreKeyBackupDialog");
+        Modal.createTrackedDialog('Restore Backup', '', RestoreKeyBackupDialog, {
+        });
+    }
+
+    _verifyDevice(e) {
+        const device = this.state.backupSigStatus.sigs[e.target.getAttribute('data-sigindex')].device;
+
+        const DeviceVerifyDialog = sdk.getComponent('views.dialogs.DeviceVerifyDialog');
+        Modal.createTrackedDialog('Device Verify Dialog', '', DeviceVerifyDialog, {
+            userId: MatrixClientPeg.get().credentials.userId,
+            device: device,
+            onFinished: () => {
+                this._loadBackupStatus();
+            },
+        });
+    }
+
+    render() {
+        const Spinner = sdk.getComponent("elements.Spinner");
+        const AccessibleButton = sdk.getComponent("elements.AccessibleButton");
+
+        if (this.state.error) {
+            return (
+                <div className="error">
+                    {_t("Unable to load key backup status")}
+                </div>
+            );
+        } else if (this.state.loading) {
+            return <Spinner />;
+        } else if (this.state.backupInfo) {
+            let clientBackupStatus;
+            if (MatrixClientPeg.get().getKeyBackupEnabled()) {
+                clientBackupStatus = _t("This device is uploading keys to this backup");
+            } else {
+                // XXX: display why and how to fix it
+                clientBackupStatus = _t(
+                    "This device is <b>not</b> uploading keys to this backup", {},
+                    {b: x => <b>{x}</b>},
+                );
+            }
+
+            let backupSigStatuses = this.state.backupSigStatus.sigs.map((sig, i) => {
+                const sigStatSub = {
```

Comment:
I'm going to assume that sigStatSub stands for signatureStatusSubstitutions, although it took a few seconds of staring at it to guess that.  I don't think it would be a disaster to spell it out fully.

Reply:
or at least sigStatSubstitutions

Reply:
i'm unconvinced that abbreviating Status to Stat really buys us much here at all, though...